openshift/CLOUD-3674.adoc

```diff
@@ -0,0 +1,90 @@
+= Unable to set datasource class / driver class at the pool level on OpenShift
+:author:            Ivo Studensky
+:email:             istudens@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+When not using a <driver ...> specification (explicit driver entries are required only when deploying drivers as modules which is a more complex scenario that is not usually necessary), it is mandatory to set the xa-datasource-class at the pool level for each <xa-datasource ...>.
+
+When not using a <driver ...> specification, for non-XA <datasource ...> pools, it is sometimes necessary (not always required since a default driver class is specified in JDBC 4 compliant implementations but this may need to be overridden in some cases) to explicitly set a <driver-class> or <datasource-class> at the pool level.
+
+No environment variables exist to support configuring the connection class at the pool level so all XA pools and some non-XA pools require custom configuration via scripts that should not be necessary in the most common cases.
+
+=== Proposed solution
+
+Introduce new environment variables which will be processed by s2i datasource scripts and which will add relevant configuration items to a datasource.
+* `<PREFIX>_DRIVER_CLASS` processed to `<driver-class>`
+* `<PREFIX>_DATASOURCE_CLASS` processed to `<datasource-class>`
+* `<PREFIX>_XA_DATASOURCE_CLASS` processed to `<xa-datasource-class>`
+where `<PREFIX>` is a prefix taken from `DATASOURCES` environment variable.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/CLOUD-3674[CLOUD-3674]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/JBEAP-19838[JBEAP-19838]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [x] QE
+
+=== Affected Projects or Components
+
+* wildfly-cekit-modules
+
+=== Other Interested Projects
+
+* wildfly-s2i
```

Comment:
Can you elaborate how this project is affected please? I suppose README.md should be updated with the new variables (documentation). Anything else?